gandalf.js

```diff
@@ -3,46 +3,49 @@
  *
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?